lib/offline/storage.js

```diff
@@ -770,6 +788,61 @@ shaka.offline.Storage.prototype.createOfflineManifest_ = function(
   };
 };
 
+/**
+ * Extract DRM initData from first segment
+ *
+ * @param {shaka.extern.Manifest} manifest
+ * @param {shaka.net.NetworkingEngine} networkingEngine
+ * @return {Promise<shaka.extern.Manifest>}
+ * @private
+ */
+shaka.offline.Storage.prototype.extractInitDataFromFirstSegment_ = async function(
```

Comment:
Since this function is to extract the init data from a segment, please do not have the network request as part of this function. Please just have it take in a segment and extract the init data from the segment.

Reply:
we could potentially break up the method for clarity; downloading the first segments and extracting the init data